src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddlewareImpl.cs

```diff
@@ -328,3 +352,10 @@ private Task DisplayRuntimeException(HttpContext context, Exception ex)
         return errorPage.ExecuteAsync(context);
     }
 }
+
+internal record ExceptionExtensionData(string Details, IHeaderDictionary Headers, string Path, string? Endpoint, RouteValueDictionary? RouteValues);
```

Comment:
For my learning:

In dotnet/runtime we typically don't use `record` because of the bloat associated with it - C# generates members you don't use/need. Do we have guidelines in dotnet/aspnetcore when to use `record` and when not? Or don't we care as much?

Reply:
I think we don't care as much. It's internal, so if we crack down on records in aspnetcore (we have them in other places) then it's simple enough to change.

Reply:
For reference, this is how much IL is generated by this single `record` line: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgH1kigBMACAUQTBgAcNsIA7KjGJgZ0aYBEBDDHwAUAAQCMABjI8YgvBzRkAkgAkYfEjCg9sYBsz5QAnmTUatCsuKkAFAQAtF1gPyUmJWhGxMMigEoQAK5sAGp8uIEwOnrchkauAcEwYREwHACUANwAsABQeXkiAExkiaHhkdH6THF5AN55AL4Fud5sUABmfDTKZprautW1uQ25jUA==

This `record` will translate to about 5kB of binary footprint in the native aot image. 80% of it is dynamically unreachable code, but it is impossible for the trimmer to prove that it is unreachable because it is all behind potentially reachable interfaces.

Reply:
Impressive! Looks like we should clean up record usage in aspnetcore.